packages/memoize/src/memoize.js

```diff
@@ -0,0 +1,28 @@
+/**
+ * Given a one-argument function `fn` or a WeakMap-key compatible
+ * argument `arg`, returns `memoFn`, a memoizing form of that function
+ * that will only call `fn(arg)` for a given `arg` the first time.
+ * The `memoFn` function remembers the result and reuses it on subsequent
+ * calls with the same `arg`.
+ *
+ * See memoize.md for the Memoization Safety properties of `memoize`.
+ * (TODO turn into link once there's a URL)
+ *
+ * @template {any} A=[any]
+ * @template {any} R=[any]
```

Comment:
If you want to constrain the type of `A` to be a WeakMap-compatible key (and unconstrain R), you can do:
```suggestion
 * @template {{}} A
 * @template R
```

Reply:
Done